lib/dep-graph-builders/yarn-lock-v1/build-depgraph-simple.ts

```diff
@@ -54,17 +57,17 @@ export const buildDepGraphYarnLockV1Simple = async (
 const dfsVisit = async (
   depGraphBuilder: DepGraphBuilder,
   node: PkgNode,
-  visitedMap: Set<string>,
   extractedYarnLockV1Pkgs: NormalisedPkgs,
   strictOutOfSync: boolean,
   includeOptionalDeps: boolean,
+  pruneWithinTopLevel: boolean,
+  visited?: Set<string>,
 ): Promise<void> => {
-  visitedMap.add(node.id);
-
   for (const [name, depInfo] of Object.entries(node.dependencies || {})) {
     if (eventLoopSpinner.isStarving()) {
       await eventLoopSpinner.spin();
     }
+    const localVisited = visited || new Set<string>();
```

Comment:
nit, but you could consider moving this out of the loop to avoid constant object creation. Not sure how bit a hit it has on the performance though.

Reply:
The reason it is in the loop is to create it on each top level iteration, after first recursion layer though it will use the `visited` passed in as a function arg to `dfsVisit`